Source/FullScan/DetailedReplication.lua

```diff
@@ -0,0 +1,42 @@
+function Auctionator.FullScan.DetailedReplication()
+  local totalReplicateItems = C_AuctionHouse.GetNumReplicateItems()
+  if totalReplicateItems==0 then
+    return --AH closed, so missing remaining data
+  end
+  local percent = math.floor(100*Auctionator.FullScan.ReplicationState.Index/totalReplicateItems)
+  if percent>Auctionator.FullScan.ReplicationState.LastPercent then
+    Auctionator.Utilities.Message("Detailed scan "..tostring(percent).."%")
+    Auctionator.FullScan.ReplicationState.LastPercent = percent
+  end
+  local notWaiting = true
```

Comment:
I really don't understand what is happening in this function?

Reply:
It is the most complicated part. As it has to keep waiting for events that supply the data needed it has to save it's state in ReplicationState. When the data is available the function is called again from the event handler and carries on with the old state, until it reaches the last item, then it adds the prices to the database.

Reply:
I can't say I like the solution, but it's was the best way I could see to handle the fact that the data isn't available immediately